src/tree.rs

```diff
@@ -355,8 +355,9 @@ impl TryFrom<Builder> for Tree {
 
         // If any parents form cycles, abort. We haven't seen cyclic trees in
         // the wild, and breaking cycles would add complexity.
-        if let Some(index) = detect_cycles(&parents) {
-            return Err(ErrorKind::Cycle(builder.entries[index].item.guid.clone()).into());
+        if let Some(index) = detect_cycles(&parents, &mut builder) {
+            let unfiled_index = builder.entry_index_by_guid[&UNFILED_GUID];
+            parents.push(ResolvedParent::ByStructure(unfiled_index));
```

Comment:
I thought ByStructure would be most applicable in this case, but I may be wrong. If I'm write I'll need to amend the assertionat line 413 that assumes in this case the parent index would equal the child's.